fbw-common/src/wasm/systems/systems/src/payload/mod.rs

```diff
@@ -0,0 +1,292 @@
+            6 => GsxState::Completed,
+            i => panic!("Cannot convert from {} to GsxState.", i),
+        }
+    }
+}
+
+read_write_enum!(GsxState);
+
+pub struct PaxInfo {
+    pub max_pax: i8,
+    pub pax_id: String,
+    pub payload_id: String,
+}
+impl PaxInfo {
+    pub fn new(max_pax: i8, pax_id: &str, payload_id: &str) -> Self {
+        PaxInfo {
+            max_pax,
+            pax_id: pax_id.to_string(),
+            payload_id: payload_id.to_string(),
+        }
+    }
+}
+
+pub struct CargoInfo {
+    pub max_cargo: Mass,
+    pub cargo_id: String,
+    pub payload_id: String,
+}
+impl CargoInfo {
+    pub fn new(max_cargo: Mass, cargo_id: &str, payload_id: &str) -> Self {
+        CargoInfo {
+            max_cargo,
+            cargo_id: cargo_id.to_string(),
+            payload_id: payload_id.to_string(),
+        }
+    }
+}
+#[derive(Debug)]
+pub struct Pax {
+    pax_id: VariableIdentifier,
+    pax_target_id: VariableIdentifier,
+    payload_id: VariableIdentifier,
+    per_pax_weight: Rc<Cell<Mass>>,
+    pax_target: u64,
+    pax: u64,
+    payload: Mass,
+}
+impl Pax {
+    pub fn new(
+        pax_id: VariableIdentifier,
+        pax_target_id: VariableIdentifier,
+        payload_id: VariableIdentifier,
+        per_pax_weight: Rc<Cell<Mass>>,
+    ) -> Self {
+        Pax {
+            pax_id,
+            pax_target_id,
+            per_pax_weight,
+            payload_id,
+            pax_target: 0,
+            pax: 0,
+            payload: Mass::new::<pound>(0.),
+        }
+    }
+
+    fn per_pax_weight(&self) -> Mass {
+        self.per_pax_weight.get()
+    }
+
+    pub fn pax_is_target(&self) -> bool {
+        self.pax == self.pax_target
+    }
+
+    pub fn pax(&self) -> u64 {
+        self.pax
+    }
+
+    pub fn pax_num(&self) -> i8 {
+        self.pax.count_ones() as i8
+    }
+
+    pub fn pax_target_num(&self) -> i8 {
+        self.pax_target.count_ones() as i8
+    }
+
+    pub fn payload(&self) -> Mass {
+        self.payload
+    }
+
+    pub fn payload_is_sync(&self) -> bool {
+        self.payload
+            == Mass::new::<pound>(self.pax_num() as f64 * self.per_pax_weight().get::<pound>())
+    }
+
+    pub fn load_payload(&mut self) {
+        self.payload =
+            Mass::new::<pound>(self.pax_num() as f64 * self.per_pax_weight().get::<pound>());
+    }
+
+    pub fn move_all_pax(&mut self) {
+        self.pax = self.pax_target;
+        self.load_payload();
+    }
+
+    pub fn move_pax(&mut self, pax: i8) {
+        for _ in 0..pax {
+            self.move_one_pax();
+        }
+    }
+
+    pub fn move_one_pax(&mut self) {
+        let pax_diff = self.pax_target_num() - self.pax_num();
+
+        let n: u64 = if pax_diff > 0 {
+            !self.pax & self.pax_target
+        } else {
+            self.pax & !self.pax_target
+        };
+        let count = n.count_ones() as i8;
+        if count > 0 {
+            let mut skip: i8 = rand::thread_rng().gen_range(0..count);
```

Comment:
You can use `random_from_range()` from systems::shared.

Reply:
Also this doesn't work in the wasm environment. `random_from_range` works.